beacon/app/app.go

```diff
@@ -243,6 +303,67 @@ func (app BeaconApp) waitForExit() error {
 	return nil
 }
 
+func (app BeaconApp) registerMessageHandlers() error {
+	err := app.registerDirectMessage("pb.GetBlockMessage", app.onMessageGetBlock)
+	if err != nil {
+		return err
+	}
+
+	err = app.registerDirectMessage("pb.BlockMessage", app.onMessageBlock)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (app BeaconApp) registerDirectMessage(messageName string, handler func(string, proto.Message)) error {
+	ctx := context.Background()
+	{
+		sub, err := app.p2pRPCClient.SubscribeDirectMessage(ctx, &pb.SubscribeDirectMessageRequest{
+			MessageName: messageName,
```

Comment:
maybe also include a peer id here so we can avoid receiving messages from peers we didn't subscribe to

Reply:
Fixed. Though I'm not sure what the use case to distinguish between peers for messages.